parser.c

```diff
@@ -0,0 +1,57 @@
+#include "minishell.h"
+
+//we have to be sure that everything is 'zero' before using the structure
+static void initialize_cmd(t_command **cmd)
+{
+    *cmd->command = NULL;
+    *cmd->path = NULL;
+    *cmd->args = NULL;
+    *cmd->pipe_output = false;
+    *cmd->pipe_fd = 0;
+    *cmd->prev = NULL;
+    *cmd->next = NULL;    
+}
+//creating new element of the list t_command
+
+t_command   *lst_new_cmd(bool value)
+{
+    t_command   *new_node;
+    
+    new_node = (t_command *)malloc(sizeof(t_command));
+    if (!new_node)
+        return (NULL);
+    ft_memset(new_node, 0, sizeof(t_command));
+    new_node->pipe_out(&new_node);
```

Comment:
`pipe_out()` is not defined - seems like it might be intended as a member of the t_command structure (e.g., a boolean or function pointer). However, the syntax `new_node->pipe_out(&new_node);` is invalid because `pipe_out` is not a callable function in the `t_command structure`
Also if we call this line here, are we sure that we use the behaviour, as we are zeroeing all variables of `new_node` on the next line?